### DISCUSSION TITLE
Enhancing Semantic Release for path-based versioning

### DISCUSSION THREAD

**Discussion by Menosc** [semantic-release/semantic-release#3112] [Q&A] [answered]
Hey everyone,

I hope you're all doing well!  I wanted to open up a discussion around a feature I believe could enhance the Semantic Release process and better cater to projects with specific directory-based versioning needs. (I couldn't find if it exists or not)

**Context**:
Currently, Semantic Release does an excellent job automating versioning based on commit messages, but I've encountered a scenario where versioning based on commit messages alone isn't sufficient. In my project, I have distinct directories (e.g., charts, docs, env, etc), and I'd love to see a way to incorporate these directory paths into the versioning process.

**Feature I'm looking for**:
I'm envisioning a feature that allows Semantic Release to consider commits based on the paths they touch. This way, I can create releases not only based on commit prefixes but also on the directories they are made in. However, I'm not certain if this feature already exists, and I need it for my monorepo setup. Basically I am looking for a way to arrange the changelog based both on commits and paths those commits are made it. 

Here's and example of a changelog structure I would like to have: 

```
1.1.0

Baseline: [Commit ID link for 1.0.0]

General

- [Name of Pull Request](PR ID link)
- [Name of Pull Request](PR ID link)
- [Name of Pull Request](PR ID link)
- [Name of Pull Request](PR ID link)

Chart 

- [Commit message](Commit ID link)
- [Commit message](Commit ID link)


Docs

- [Commit message](Commit ID link)
- [Commit message](Commit ID link)

Envs

- [Commit message](Commit ID link)
- [Commit message](Commit ID link)

```

Thank you in advance!

**Comment by travi** [comment, marked as answer]
Sorry, this is not in line with the goals of the semantic-release project. 

How folks structure projects is far too varied for us to attempt to enable this with our small maintenance team, even if we were to agree that this was a direction we wanted to follow. 

Semantic-release expects the committer to translate details like you suggest to match the chosen commit convention.
